src/gui.cpp

```diff
@@ -1798,6 +1798,44 @@ std::string DDOPGeneratorGUI::get_object_type_string(isobus::task_controller_obj
 	return retVal;
 }
 
+std::string DDOPGeneratorGUI::get_object_display_name(std::shared_ptr<isobus::task_controller_object::Object> object)
+{
+	std::string displayName = object->get_designator();
+	
+	// If designator is empty or is the default "Designator" text, use DDI name for DPD and DPT objects
+	const auto objectType = object->get_object_type();
+	const bool isDPDorDPT = (objectType == isobus::task_controller_object::ObjectTypes::DeviceProcessData) ||
+	                        (objectType == isobus::task_controller_object::ObjectTypes::DeviceProperty);
+	
+	if ((displayName.empty() || displayName == "Designator") && isDPDorDPT)
+	{
+		std::uint16_t ddi = 0;
+		if (objectType == isobus::task_controller_object::ObjectTypes::DeviceProcessData)
+		{
+			auto dpd = std::dynamic_pointer_cast<isobus::task_controller_object::DeviceProcessDataObject>(object);
+			if (dpd != nullptr)
+			{
+				ddi = dpd->get_ddi();
+			}
+		}
+		else // DeviceProperty
+		{
+			auto dpt = std::dynamic_pointer_cast<isobus::task_controller_object::DevicePropertyObject>(object);
+			if (dpt != nullptr)
+			{
+				ddi = dpt->get_ddi();
+			}
+		}
+		
+		if (ddi != 0)
+		{
+			displayName = isobus::DataDictionary::get_entry(ddi).name;
```

Comment:
Lets just do this twice where the ddi is assigned. Then we can remove the outer if check and just check object type once. Also remove ddi variable